tests/Console/ConfigurationResolverTest.php

```diff
@@ -961,6 +961,26 @@ public function provideResolveRulesCases()
                     '@A' => true,
                 ),
             ),
+            '@A - @C' => array(
```

Comment:
>```
> @Foo => { 'A' => 'foo'}
> @Bar => { 'A' => 'bar'}
> ```
> then `@Foo - @Bar = {}` , which currently resolves to `{'A' => 'foo'}`

This feels really weird to me. It's like subtracting an apple from an orange. I'd go with  the current resolution: `A` would only be removed if values match. For example:
```
@Foo => { 'A' => 'foobar'}
@Bar => { 'A' => 'foobar'}
@Foo - @Bar => {}
```
But
```
@Foo => { 'A' => 'foobar'}
@Bar => { 'A' => 'bar'}
@Foo - @Bar => { 'A' => 'foobar'}
```
Since it wouldn't be able to find a `'A' => 'bar'` to remove.
```
@Foo => { 'A' => ['foo', 'bar']}
@Bar => { 'A' => ['bar']}
@Foo - @Bar => { 'A' => ['foo']}
```
Would make perfect sense, though.

And for me there is a difference between `@Foo - @Bar` and `@Foo = true, @Bar = false`. The first removes from `@Foo` all rules set to true in `@Bar` (consider as if a rule set to false in `@Bar` doesn't exist). The later applies all rules defined in `@Foo` **and then** sets all rules present in `@Bar` as false. The first is an operation while the second is an assignment.

To minimize confusion, I believe it would be better to have only assignments and no subtractions. Unless there're some use cases in which ruleSet subtraction would make so much sense it could justify the whole mess. :man_shrugging: